### PR TITLE
Use Redis keyPrefix over db

### PR DESCRIPTION
**Pull request by dariacm**
This PR replaces usage of Redis `db` with `keyPrefix`. This is not only the desired approach by our Platform team, but also the one recommended by Redis creators themselves (https://groups.google.com/g/redis-db/c/vS5wX8X4Cjg/m/8ounBXitG4sJ?pli=1).

### CODE REVIEW

**Comment by drdaemos** [comment]
Btw, `keyPrefix` is not compatible with BullMQ: https://docs.bullmq.io/guide/connections

**Comment by dariacm** [comment]
> Btw, `keyPrefix` is not compatible with BullMQ: https://docs.bullmq.io/guide/connections

Yeah 💀 We changed our own BullMQ implementation to remove the prefix from the Redis configuration and use it to fill the BullMQ ad-hoc field for it in https://github.com/lokalise/shared-ts-libs/pull/225

File: src/infrastructure/config.ts [hunk 1]
```diff
@@ -153,7 +153,7 @@ export function getDbConfig(): DbConfig {
 export function getRedisConfig(): RedisConfig {
   return {
     host: configScope.getMandatory('REDIS_HOST'),
-    db: configScope.getMandatoryValidatedInteger('REDIS_DB', redisDbValidator),
```
[the remark this answers is not in the document]
Comment (dariacm) [reply src/infrastructure/config.ts:156]: Hmm good catch, it's used for the scheduler DB but I assume we want to make the change there as well

[the remark this answers is not in the document]
Comment (kibertoad) [reply src/infrastructure/config.ts:156]: ya

**Pull request merged**